### PR TITLE
Domain designer cleanup for consistent use of dirty state check with BeforeUnload

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Recent domain designer stories included a refactor of the dirty state handling for the component into a shared `<BeforeUnload>` helper. This PR folds that into some of the other domain designer wrapper classes for consistency. Also included in this PR is a few misc list designer code clean-up items.

#### Changes
* Domain designer cleanup for consistent use of dirty state check with BeforeUnload
* Remove unused ListImportServiceImpl.java
* Revert addition of the GWTPropertyDescriptor disablePhiLevel member variable since it is now being handled on the ui-components API post side
* ListDomainKindProperties member variables should be private

### CODE REVIEW

File: query/src/client/QueryMetadataEditor/QueryMetadataEditor.tsx [hunk 1]
```diff
@@ -51,6 +51,8 @@ interface IAppState {
 
 export class App extends PureComponent<any, Partial<IAppState>> {
 
+    private _dirty = false;
```
Comment (labkey-ankurj) [inline query/src/client/QueryMetadataEditor/QueryMetadataEditor.tsx]:
```suggestion
    private _dirty: boolean = false;
```

File: assay/src/client/AssayDesigner/AssayDesigner.tsx [hunk 2]
```diff
@@ -29,11 +29,12 @@ type State = {
     isLoadingModel: boolean,
     hasDesignAssayPerm?: boolean
     message?: string
-    dirty: boolean
 }
 
 export class App extends React.Component<any, State> {
 
+    private _dirty = false;
```
Comment (labkey-ankurj) [inline assay/src/client/AssayDesigner/AssayDesigner.tsx]: Should _dirty be defined of type boolean?
```suggestion
    private _dirty: boolean = false;
```

File: experiment/src/client/DomainDesigner/DomainDesigner.tsx [hunk 3]
```diff
@@ -38,31 +42,27 @@ interface IAppState {
 
 export class App extends React.PureComponent<any, Partial<IAppState>> {
 
+    private _dirty = false;
```
Comment (labkey-ankurj) [inline experiment/src/client/DomainDesigner/DomainDesigner.tsx]:
```suggestion
    private _dirty: boolean = false;
```

File: list/src/client/Designer/Designer.tsx [hunk 4]
```diff
@@ -23,40 +23,35 @@ import {
...
-    dirty: boolean,
     model?: ListModel
     fileImportError: string
 }
 
-export class App extends React.Component<{}, State>
-{
+export class App extends React.Component<{}, State> {
+
+    private _dirty = false;
```
Comment (labkey-ankurj) [inline list/src/client/Designer/Designer.tsx]:
```suggestion
    private _dirty: boolean = false;
```

File: study/src/client/DatasetDesigner/DatasetDesigner.tsx [hunk 5]
```diff
@@ -15,62 +15,49 @@
...
     isLoadingModel: boolean,
     message?: string,
-    dirty: boolean,
-    returnUrl: string,
     fileImportError: string
 }
 
 export class App extends PureComponent<any, State> {
+
+    private _dirty = false;
```
Comment (labkey-ankurj) [inline study/src/client/DatasetDesigner/DatasetDesigner.tsx]:
```suggestion
    private _dirty: boolean = false;
```

File: study/src/client/DatasetDesigner/DatasetDesigner.tsx [hunk 6]
```diff
@@ -91,11 +78,10 @@ export class App extends PureComponent<any, State> {
     }
 
     navigate(defaultUrl: string) {
-        const { returnUrl } = this.state;
+        this._dirty = false;
```
Comment (labkey-ankurj) [inline study/src/client/DatasetDesigner/DatasetDesigner.tsx:81]: is 'this._dirty' accessible in this method as it is not binding with () => .. ? I am not sure though.

**Review by labkey-ankurj — APPROVED** [review]
minor edits and 1 question

File: study/src/client/DatasetDesigner/DatasetDesigner.tsx (hunk 6 above)
Comment (cnathe) [reply study/src/client/DatasetDesigner/DatasetDesigner.tsx:81]: Yes, the scoping for `this` should be good to go here. I will verify just to be sure.

**Pull request merged**